### PR TITLE
Editorconfig & Format chapter number to %03d 

### PR DESCRIPTION
**Pull request by kuanyui**
For #22 
Sorry for late. Too busy recently.

### CODE REVIEW

**Comment by eight04** [comment]
I am going to make an option for this.

**Comment by eight04** [comment]
To apply number format, add following to setting.ini:
```
[動漫屋]
titlenumberformat = {:03d}
```

**Comment by kuanyui** [comment]
Where is the RegExp pattern `第(\d+)[話话卷]`?

File: comiccrawler/core/__init__.py [hunk 1]
```diff
@@ -672,3 +680,10 @@ def analyze_info(mission, mod):
 	remove_duplicate_episode(mission)
 
 	print("Analyzing success!")
+
+def format_number(title, format):
+	"""第3卷 --> 第003卷"""
+	def replacer(match):
+		number = match.group()
+		return format.format(int(number))
+	return re.sub("\d+", replacer, title)
```
Comment (kuanyui) [inline comiccrawler/core/__init__.py]: I think this may cause wrongly replacement when the title itself contains digits.

Comment (eight04) [inline comiccrawler/core/__init__.py]: Maybe we should try to trim the mission title from the episode title first?

Some title examples:
```
泰安路47号 第006回
http://www.dm5.com/manhua-taianlu-47-hao/

原作版108(7)
第28-29話
http://tw.ikanman.com/comic/9637/
```

Comment (kuanyui) [inline comiccrawler/core/__init__.py]: Every site has different rules, so I use `第(\d+)[話话卷]?` such a conservative pattern. Or maybe each site has its digits formatter?

Comment (eight04) [inline comiccrawler/core/__init__.py]: Adding new API to format title is the safest, but let's consider if there is an universal solution first.

With above solution,
```
泰安路47号 第006回
原作版108(7)
第28-29話
```
becomes
```
泰安路47号 第006回
原作版108(007)
第028-029話
```
which looks good to me.

Comment (kuanyui) [inline comiccrawler/core/__init__.py]: Take a look of this extreme example: http://www.dm5.com/manhua-bianchengnageta/

Comment (eight04) [inline comiccrawler/core/__init__.py]: I didn't see the problem.

Comment (kuanyui) [inline comiccrawler/core/__init__.py]: Sorry, my fault. I misunderstood something.

**Comment by kuanyui** [comment]
Awesome! I think it's time to merge this PR.

BTW, Your skill in Python is so good, I never know that`re.sub()` can accept a function as parameter (Seldom see functional programming in Python, though I think that's due to my poor experience...)

**Comment by eight04** [comment]
Released in 2017.1.6.

**Pull request merged**